test-e2e/config-loading.test.ts

```diff
@@ -0,0 +1,48 @@
+import { testSubdomainRouting as test, expect } from './fixtures/config-test-fixtures.js'
+import { getConfig, setConfig } from './fixtures/set-sw-config.js'
+import { waitForServiceWorker } from './fixtures/wait-for-service-worker.js'
+import type { ConfigDb } from '../src/lib/config-db'
+
+test.describe('/#/ipfs-sw-config', () => {
+  const testConfig: ConfigDb = {
+    gateways: [process.env.KUBO_GATEWAY as string, 'http://example.com'],
+    routers: [process.env.KUBO_GATEWAY as string, 'http://example.com/routing/v1'],
+    dnsJsonResolvers: {
+      '.': 'example.com/dns-query'
+    },
+    debug: 'testDebug',
+    enableWss: false,
+    enableWebTransport: true,
+    enableRecursiveGateways: false,
+    enableGatewayProviders: false
+  }
```

Comment:
I attempted to set non-default values here just so we can be sure it's overriding what would be set without any changes